src/delegate/delegate.ios.ts

```diff
@@ -259,10 +258,35 @@ export class TwilioAppDelegate extends UIResponder
       }
   }
 
-  reportIncomingCall(from: String, uuid: any) {
+  reportIncomingCall(from: String, uuid: any, callInvite : TVOCallInvite) {
+    // let customParameters: NSDictionary <string,string> = callInvite.customParameters;
+    
+    // if(customParameters) {
+    //   console.debug("Custom Parameters Detected:",customParameters);
+    // }
+
+    console.log('richard:',callInvite.customParameters);
+
+    // Parse NSDictionary into JSON Object
+    var jsonData : any = NSJSONSerialization.dataWithJSONObjectOptionsError(callInvite.customParameters, 0, null); 
+
+    // Parse the JSON Data into a JSONObject
+    var JSONObject : Object = JSON.parse(NSString.alloc().initWithDataEncoding(jsonData, 4).toString());
```

Comment:
I would change this fragment to:
```
    // Parse NSDictionary into JSON Object
    const jsonData : any = NSJSONSerialization.dataWithJSONObjectOptionsError(callInvite.customParameters, 0, null); 

    const jsonObject : Object = JSON.parse(NSString.alloc().initWithDataEncoding(jsonData, 4).toString());
```

But, BTW. Why are you processing all of that? I think you are using only the `from` attribute, right?

Reply:
Agreed I added new comments to that part and assigned them as const declaration instead of var 💯 

Oh also, the reason that i parsed it this way is because users can actually send other custom data as well, and actually this is a function that we are looking into for our own platform as well. for example sometimes we will send the data as subscriber_whole_name or group_call_name (for conference calls)

that way the user would be able to call them as param.subscriber_whole_name or param.group_call_name 💯 

but we can restrict the custom data to a certain field if you think its better for example, we can add like a customInput array which the data will parse from.

but actually when i looked inside of the callInvite.customParameters it only contains the Custom Call Parameter data set by the user so in my case it only contained {name="richard"} which i thought was okay to parse since it will only contain the relevant data , twilio doesnt send any extra params into this function